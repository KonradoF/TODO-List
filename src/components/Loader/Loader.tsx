import React, {ReactElement} from 'react';
import {ActivityIndicator} from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';

interface Props {
  secondary?: boolean;
}
export const Loader = ({secondary}: Props): ReactElement => {
  return (
    <ActivityIndicator
      color={
        secondary
          ? EStyleSheet.value('$loaderSecondaryColor')
          : EStyleSheet.value('$loaderDefaultColor')
      }
      size="large"
    />
  );
};
