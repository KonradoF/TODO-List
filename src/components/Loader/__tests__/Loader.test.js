import React from 'react';
import {shallow} from 'enzyme';
import {Loader} from '../Loader';

describe('Loader', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<Loader />);
      expect(component).toMatchSnapshot();
    });
    it('should match to snapshot - secondary', () => {
      const component = shallow(<Loader secondary />);
      expect(component).toMatchSnapshot();
    });
  });
});
