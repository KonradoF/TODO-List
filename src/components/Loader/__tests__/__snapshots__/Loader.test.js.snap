// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Rendering should match to snapshot - secondary 1`] = `
<ActivityIndicator
  animating={true}
  color="#ffffff"
  hidesWhenStopped={true}
  size="large"
/>
`;

exports[`Loader Rendering should match to snapshot 1`] = `
<ActivityIndicator
  animating={true}
  color="#F36A08"
  hidesWhenStopped={true}
  size="large"
/>
`;
